### PR TITLE
feat: add crate that allows printing the name of an enum variant

### PR DESCRIPTION
**Pull request by huitseeker**
Used for logging cool stuff

### CODE REVIEW

File: name_variant/examples/test.rs [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    assert_eq!(x.variant_name(), "A");
+
+    let x = TestEnum::B();
+    assert_eq!(x.variant_name(), "B");
+
+    let x = TestEnum::D {
+        _name: "Jane Doe".into(),
+        _age: 30,
+    };
+    assert_eq!(x.variant_name(), "D");
```
Comment (velvia) [inline name_variant/examples/test.rs]: Could we add something that uses x after the assert so we can be sure that variant_name() does not move x and allows x to be used later?

File: name_variant/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+    };
+}
+
+fn match_enum_to_string(name: &Ident, variants: &DataEnum) -> proc_macro2::TokenStream {
+    // the variant dispatch proper
+    let mut match_arms = quote! {};
+    for variant in variants.variants.iter() {
+        let variant_ident = &variant.ident;
+        let fields_in_variant = match &variant.fields {
+            Fields::Unnamed(_) => quote_spanned! {variant.span()=> (..) },
```
Comment (velvia) [inline name_variant/src/lib.rs]: space before `=>` ?

**Review by velvia — APPROVED** [review]
Thanks @huitseeker .... this looks awesome!  Much simpler than strum.

File: name_variant/examples/test.rs (hunk 1 above)
Comment (huitseeker) [reply name_variant/examples/test.rs]: I'm now forcing the receiver, using `(&x).variant_name()` syntax.

**Pull request merged**